packages/dev/core/src/Materials/PBR/openPbrMaterial.ts

```diff
@@ -0,0 +1,2550 @@
+                        ubo.updateVector3(uniform.name, TmpVectors.Vector3[0]);
+                    } else if (uniform.numComponents === 2) {
+                        uniform.populateVectorFromLinkedProperties(TmpVectors.Vector2[0]);
+                        ubo.updateFloat2(uniform.name, TmpVectors.Vector2[0].x, TmpVectors.Vector2[0].y);
+                    } else if (uniform.numComponents === 1) {
+                        ubo.updateFloat(uniform.name, uniform.linkedProperties[Object.keys(uniform.linkedProperties)[0]].value);
+                    }
+                });
+
+                // Misc
+                this._lightingInfos.x = this._directIntensity;
+                this._lightingInfos.y = this.emissionLuminance;
+                this._lightingInfos.z = this._environmentIntensity * scene.environmentIntensity;
+                this._lightingInfos.w = 1.0; // This is used to be _specularIntensity.
+
+                ubo.updateVector4("vLightingIntensity", this._lightingInfos);
+
+                ubo.updateFloat2("vDebugMode", this.debugLimit, this.debugFactor);
+            }
+
+            // Textures
+            if (scene.texturesEnabled) {
+                // Loop through samplers and set textures
+                for (const key in this._samplersList) {
+                    const sampler = this._samplersList[key];
+                    if (sampler.value) {
+                        ubo.setTexture(sampler.samplerName, sampler.value);
+                    }
+                }
+
+                BindIBLSamplers(scene, defines, ubo, radianceTexture, this.realTimeFiltering);
+
+                if (defines.ENVIRONMENTBRDF) {
+                    ubo.setTexture("environmentBrdfSampler", this._environmentBRDFTexture);
+                }
+
+                if (defines.ANISOTROPIC) {
+                    ubo.setTexture("blueNoiseSampler", OpenPBRMaterial._noiseTexture);
+                }
+            }
+
+            // OIT with depth peeling
+            if (this.getScene().useOrderIndependentTransparency && this.needAlphaBlendingForMesh(mesh)) {
+                this.getScene().depthPeelingRenderer!.bind(effect);
+            }
+
+            this._eventInfo.subMesh = subMesh;
+            this._callbackPluginEventBindForSubMesh(this._eventInfo);
+
+            // Clip plane
+            BindClipPlane(this._activeEffect, this, scene);
+
+            this.bindEyePosition(effect);
+        } else if (scene.getEngine()._features.needToAlwaysBindUniformBuffers) {
+            this._needToBindSceneUbo = true;
+        }
+
+        if (mustRebind || !this.isFrozen) {
+            // Lights
+            if (scene.lightsEnabled && !this._disableLighting) {
+                BindLights(scene, mesh, this._activeEffect, defines, this._maxSimultaneousLights);
+            }
+
+            // View
+            if ((scene.fogEnabled && mesh.applyFog && scene.fogMode !== Scene.FOGMODE_NONE) || radianceTexture || mesh.receiveShadows || defines.PREPASS) {

```

Comment:
We have some new code here related to clustered lights (see **PBRBaseMaterial.ts**).

Reply:
Yup, it's on my list of things to do.